nntrainer/layers/split_layer.cpp

```diff
@@ -92,17 +92,38 @@ int SplitLayer::initialize(Manager &manager) {
   output_reshape_helper = input_reshape_helper;
   output_reshape_helper.height(1);
 
+  std::cout << output_reshape_helper;
+
   return status;
 }
 
+/**
+ * @brief create a helper dimension with a new batch set
+ *
+ * @param batch
+ * @param base
+ * @return TensorDim
+ */
+static TensorDim setHelperBatch(unsigned int batch, const TensorDim &base) {
```

Comment:
how about keep L76 the original version, and rather override setBatch() function here to update the input_reshape_helper/output_reshape_helper (will be a simple scale).
this will also keep rest of the code same and no overhead in fowarding/backwarding.

Reply:
That was my first idea. That cannot be done because, in that way, setBatch has side effect.

consecutive call of 
setBatch(20);
setBatch(20);

will accumulate and there seems no way to know if we have to reset the batch or the batch has information of split dimensions.

Actually we can have another member to save this state. I'll make a tweak

Reply:
Updated.

Reply:
looks good now.